providers/amazon/src/airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -603,12 +607,22 @@ def conn_config(self) -> AwsConnectionWrapper:
         """Get the Airflow Connection object and wrap it in helper (cached)."""
         connection = None
         if self.aws_conn_id:
+            possible_exceptions = (
+                (AirflowNotFoundException, AirflowRuntimeError)
```

Comment:
I'm curious how you're still getting the `AirflowRuntimeError` even after the change in #47648

Reply:
@ramitkataria the change you implemented is in `airflow.models.connection.Connection` which is backwards-compat way to get Connection. The new way is from `airflow.sdk`:

https://github.com/apache/airflow/blob/7c292141cb69ee37c5b2bd621cae85526b828b03/task-sdk/src/airflow/sdk/definitions/connection.py#L138-L142

https://github.com/apache/airflow/blob/7c292141cb69ee37c5b2bd621cae85526b828b03/task-sdk/src/airflow/sdk/execution_time/context.py#L156-L161

Reply:
I see, thanks for the clarification! Do you think we should do something similar to this in `context.py` as well for consistency?
https://github.com/apache/airflow/blob/3dc597f02adec99cec282790c0575b15dc8f0f3f/airflow-core/src/airflow/models/connection.py#L479-L482
It might help features from breaking in other providers as well

Reply:
Can we resolve this conversation?

Reply:
I think it's a valid discussion, but we shouldn't be blocking this PR for it.    Unless I am missing something and there is something else I need to implement here?

Reply:
Yea I think we can resolve and merge this for now and discuss on slack instead? If we end up going with the approach I proposed, then the changes in this PR wouldn't be needed anymore and instead, the change would be in sdk code